sparse_ho/tests/test_models.py

```diff
@@ -88,6 +89,37 @@ def test_beta_jac_custom(model_name):
         assert np.allclose(jac, jac_custom)
 
 
+@pytest.mark.parametrize('model_name', list(custom_models.keys()))
+def test_warm_start(model_name):
+    """Check that using sk or celer yields the same solution as sparse ho"""
+    if model_name in ("svm", "svr", "ssvr"):
+        X_s = X_r
+    else:
+        X_s = X_c
+    model = models[model_name]
+
+    for log_alpha in dict_list_log_alphas[model_name]:
+        mask, dense, jac = None, None, None
+        for i in range(2):
+            mask, dense, _ = compute_beta(
+                X_s, y, log_alpha, tol=tol,
+                mask0=mask, dense0=dense, jac0=jac,
+                max_iter=5000, compute_jac=False, model=model)
+            dbeta0_new = model._init_dbeta0(mask, mask, jac)
+            reduce_alpha = model._reduce_alpha(np.exp(log_alpha), mask)
+
+            _, dual_var = model._init_beta_dual_var(X_s, y, mask, dense)
+            jac = get_only_jac(
+                model.reduce_X(X_s, mask), model.reduce_y(y, mask), dual_var,
+                reduce_alpha, model.sign(dense, log_alpha), dbeta=dbeta0_new,
+                niter_jac=5000, tol_jac=1e-13, model=model, mask=mask,
+                dense=dense)
+            if i == 0:
+                assert get_only_jac.n_iter > 2
```

Comment:
use np.testing.assert_array_less as it prints the value if it fails (while assert just says that the assertion failed and makes debugging one step harder)

Reply:
I don't understand get_only_jac.n_iter ? this works ?

Reply:
yes function in python are objects where you can define atributes. 
I don't know what's best between that of the 
if :
 return _, _
else
return _